docs/pin.rst

```diff
@@ -0,0 +1,112 @@
+it. There are 19 pins for your disposal, numbered 0-16 and 19-20. Pins 17 and
+18 are not available.
+
+.. image:: pinout.png
+
+Those pins are available as attributes on the ``microbit``
+module:``microbit.pin0`` - ``microbit.pin20``.
+
++-----+---------+----------+
+| Pin | Type    | Function |
++=====+=========+==========+
+|  0  | Touch   | Pad 0    |
++-----+---------+----------+
+|  1  | Touch   | Pad 1    |
++-----+---------+----------+
+|  2  | Touch   | Pad 2    |
++-----+---------+----------+
+|  3  | Analog  | Column 1 |
++-----+---------+----------+
+|  4  | Digital | Button A |
++-----+---------+----------+
+|  5  | Analog  | Column 2 |
++-----+---------+----------+
+|  6  | Digital | Row 2    |
++-----+---------+----------+
+|  7  | Digital | Row 1    |
++-----+---------+----------+
+|  8  | Digital |          |
++-----+---------+----------+
+|  9  | Digital | Row 3    |
++-----+---------+----------+
+|  10 | Analog  | Column 3 |
++-----+---------+----------+
+|  11 | Digital | Button B |
++-----+---------+----------+
+|  12 | Digital |          |
++-----+---------+----------+
+|  13 | Digital | SPI MOSI |
++-----+---------+----------+
+|  14 | Digital | SPI MISO |
++-----+---------+----------+
+|  15 | Digital | SPI SCK  |
++-----+---------+----------+
+|  16 | Digital |          |
++-----+---------+----------+
++-----+---------+----------+
+|  19 | Digital | I2C SCL  |
++-----+---------+----------+
+|  20 | Digital | I2C SDA  |
++-----+---------+----------+
+
+The above table summarizes the pins available, their types (see below) and what
+they are internally connected to.
+
+
+Classes
+=======
+
+There are three kinds of pins, differing in what is available for them. They
+are represented by the below classes. Note that they form a hierarchy, so that
+each class has all the functionality of the previous class, and adds its own
+to that.
+
+.. note::
+    Those classes are not actually available for the user, you can't create
+    new instances of them. You can only use the instaces already provided,
+    representing the physical pins on your board.
+
+.. py:class:: MicroBitDigitalPin
+
+    .. py:method:: read_digital()
+
+        Return 1 if the pin is high, and 0 if it's low.
+
+    .. py:method:: write_digital(value)
+
+        Set the pin to high if ``value`` is 1, or to low, if it is 0.
+
+.. py:class:: MicroBitAnalogDigitalPin(MicroBitDigitalPin)
+
+    .. py:method:: read_analog()
+
+        Read the voltage applied to the pin, and return it as an integer
+        between 0 (meaning 0V) and 1023 (meaning 3.3V).
+
+    .. py:method:: write_analog(value)
+
+        Output a PWM signal on the pin, with the duty cycle proportional to
+        the provided ``value``. The ``value`` may be either an integer or a
+        floating point number between 0 (0% duty cycle) and 1023 (100% duty).
+
+    .. py:method:: set_analog_period(period)
+
+        Set the period of the PWM signal being output to ``period``.
```

Comment:
Unit is milliseconds.  Allowed range is hard to figure out.  It will need testing on actual hardware.